### ISSUE TITLE
Add FPN and other backbone networks like ResneXt

### ISSUE THREAD

**Issue by manoja328** [jwyang/faster-rcnn.pytorch#122]
Are you thinking of adding more backbone networks like ResneXt or FPN etc. Thanks.

**Comment by jwyang** [comment]
Hi, @manoja328 , please check readme, there is a pointer to our FPN implementation, but it does not work as well as the model provided by detectron.

**Comment by Karthik-Suresh93** [comment]
Hi, can you tell me what changes I have to make in resnet.py to be able load resnext model? I have the resnext.pth and resnext.py files

**Comment by Roujack** [comment]
Another repo confirm that a few modification on fpn.pytorch repo will make it work:https://github.com/guoruoqian/FPN_Pytorch

**Comment by americanexplorer13** [comment]
> Hi, can you tell me what changes I have to make in resnet.py to be able load resnext model? I have the resnext.pth and resnext.py files

Hey, I think I'm bit late, but for other people who would find this thread:
Actually, you can do this in torchvision library. All you need is to redefine faster-rcnn model's backbone and wrap it to _resnet_fpn_extractor, i.e:

FASTER_RCNN.backbone = resnext50_32x4d(weights="ResNeXt50_32X4D_Weights.IMAGENET1K_V1", progress=True)

FASTER_RCNN.backbone = _resnet_fpn_extractor(FASTER_RCNN.backbone, 5, norm_layer=nn.BatchNorm2d)

_resnet_fpn_extractor already has all functionality to work with different resnet-like architectures (see code to find out more), I wonder why they didn't add them by default...
